forge/src/trace/mod.rs

```diff
@@ -0,0 +1,371 @@
+/// Call trace address identifiers.
+///
+/// Identifiers figure out what ABIs and labels belong to all the addresses of the trace.
+pub mod identifier;
+pub use identifier::TraceIdentifier;
+
+mod decoder;
+pub use decoder::CallTraceDecoder;
+
+use crate::abi::CHEATCODE_ADDRESS;
+use ansi_term::Colour;
+use ethers::{
+    abi::{Address, RawLog},
+    types::U256,
+};
+use serde::{Deserialize, Serialize};
+use std::fmt::{self, Write};
+
+/// An arena of [CallTraceNode]s
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct CallTraceArena {
```

Comment:
This file is also pretty much the same except I simplified it as much as I could by removing a bunch of stuff or pulling it into other abstractions. It only collects and displays traces now, and displaying is not done by calling `construct_trace_string(...)`, it just uses `Display`

Reply:
yep lg